src/utilities/oas-validator.ts

```diff
@@ -152,66 +224,84 @@ class OASValidator {
         failures.push(new EnumMismatch(path, enumValues, actual));
       }
     }
+    return failures;
+  }
+
+  private static checkItemSchemaErrors(
+    actual: Json,
+    expected: SchemaObject,
+    path: string[],
+  ): ValidationFailure[] {
+    let failures: ValidationFailure[] = [];
+    // check that the expected object's items property is set
+    const itemSchema = expected.items;
+    if (itemSchema) {
+      // re-run for each item
+      actual.forEach((item) => {
+        failures = [
+          ...failures,
+          ...this.validateObjectAgainstSchema(item, itemSchema, path),
+        ];
+      });
+    } else {
+      failures = [...failures, new InvalidSchema(ITEMS_MISSING_ERROR, path)];
```

Comment:
I think the failures array will be empty if this statement is hit. Is the `...failures` needed?